src/TorchSharp/TorchVision/Image.cs

```diff
@@ -0,0 +1,177 @@
+using System.IO;
+using System.Threading.Tasks;
+
+using static TorchSharp.torch;
+
+namespace TorchSharp.torchvision
+{
+    public static partial class io
+    {
+        /// <summary>
+        /// <cref>Imager</cref> to be used when a <cref>torchvision.io</cref> image method's <c>imager</c> is unspecified.
+        /// </summary>
+        public static Imager DefaultImager { get; set; }
+
+        /// <summary>
+        /// Abstract class providing a generic way to decode and encode images as <cref>Tensor</cref>s.
+        /// Used by <cref>torchvision.io</cref> image methods.
+        /// </summary>
+        public abstract class Imager
+        {
+            /// <summary>
+            /// Reads the contents of an image file and return the image's file format.
+            /// </summary>
+            /// <param name="image">Image file contents.</param>
+            /// <returns>The detected image format.</returns>
+            public abstract ImageFormat DetectFormat(byte[] image);
+            /// <summary>
+            /// Reads the contents of an image file and returns the result as a <cref>Tensor</cref>.
+            /// </summary>
+            /// <param name="image">Image file contents.</param>
+            /// <param name="mode">Image read mode.</param>
+            /// <remarks>
+            /// The image format is detected from image file contents.
+            /// </remarks>
+            /// <returns>
+            /// <cref>Tensor</cref> with <c>shape = [color_channels, image_height, image_width]</c> and <c>dtype = uint8</c>.
+            /// </returns>
+            public Tensor DecodeImage(byte[] image, ImageReadMode mode = ImageReadMode.UNCHANGED)
```

Comment:
I think it makes sense to add an overload that takes a stream, in addition to the one taking a byte[].